rust-runtime/aws-smithy-http-server/src/plugin/filter.rs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use tower::util::Either;
+
+use crate::operation::{Operation, OperationShape};
+
+use super::Plugin;
+
+/// A [`Plugin`] used to filter [`Plugin::map`] application using a predicate over the [`OperationShape::NAME`].
+///
+/// See [`PluginExt::filter_by_operation_name`](super::PluginExt::filter_by_operation_name) for more information.
+pub struct FilterByOperationName<Inner, F> {
+    inner: Inner,
+    predicate: F,
+}
+
+impl<Inner, F> FilterByOperationName<Inner, F> {
+    /// Creates a new [`FilterByOperationName`].
+    pub fn new(inner: Inner, predicate: F) -> Self {
+        Self { inner, predicate }
+    }
+}
+
+impl<P, Op, S, L, Inner, F> Plugin<P, Op, S, L> for FilterByOperationName<Inner, F>
+where
+    F: Fn(&'static str) -> bool,
```

Comment:
Instead of matching on a string, wouldn't it be better (i.e, more type-safe) to generate an `enum` of available operations and use that to filter?
i.e,
```rust
// Current 
let filtered_plugin = plugin.filter_by_operation_name(|name| name != "com.aws.example#CheckHealth");

// vs

let filtered_plugin = plugin.filter_by_operation(|op| op != Operation::CheckHealth);
```

Granted, the "magic strings" issue is partially solved if you use things like `CheckHealth::NAME`.

Reply:
Good idea
https://github.com/awslabs/smithy-rs/issues/1839